server/service/src/stocktake/update.rs

```diff
@@ -241,15 +250,16 @@ fn generate_stock_line_update(
         barcode_id: stock_line.barcode_id.clone(),
     };
 
-    let item = match ItemRowRepository::new(connection).find_one_by_id(&stock_line.item_id)? {
-        Some(item) => item,
-        None => {
-            return Err(UpdateStocktakeError::InternalError(format!(
-                "Can't find item {} for existing stocktake line {}!",
-                &stock_line.item_id, stocktake_line.line.id
-            )))
-        }
-    };
+    let stock_line_item =
+        match ItemRowRepository::new(connection).find_one_by_id(&stock_line_item_id)? {
```

Comment:
The stocktake load the stock_line_row which doesn't has the stock_line item loaded...

Here raises the question: who load who and where to stop loading? The problem would be even worth if there are circular table references.

In domain driven design, you would define an aggregate root, e.g. stocktake or stock_line. All roots are allowed to only contain references to other roots (which avoids the circular reference problem). So far we were fine to just load what we needed, but we might have to have a look at some of these DDD ideas...